src/sqlfluff/core/rules/base.py

```diff
@@ -503,12 +520,12 @@ class BaseRule:
     # Lint loop / crawl behavior. When appropriate, rules can (and should)
     # override these values to make linting faster.
     recurse_into = True
-    needs_raw_stack = True
+    needs_raw_stack = False  # False is faster & most rules don't need it
```

Comment:
Can we give more commentary as towhy a rule would need or not need `raw_stack`? I presume it's if you are looking at a segment's parents to find if it's nested in a particular statement? Remind me again what the difference is. between `raw_stack` and `raw_segment_pre`

Reply:
I'll add more comments`.  Briefly,`raw_segment_pre` is the **one** raw segment prior to the current one, while `raw_stack` is a list of **every** raw segment prior. The former is pretty lightweight to compute, while the latter is pretty heavyweight, as we're creating a new (and increasingly lengthy) tuple as we proceed through the file.

Reply:
Oh I didn’t get that.

Is that just the parent Segment? Would that be a better name for it rather than `raw_stack_prev`?

Reply:
No, it's not the parent. Raw segments are (IIRC) those at the leaf of the tree, i.e. no children. Generally, these are a single keyword, operator, literal, or a meta segment like indent or dedent, like the "skin" of the tree as opposed to the "bones", I guess?

Reply:
`previous_raw_segement` seems better as a name or `raw_segment_previous`.
at least `prev`